src/test/java/storage/MangaDeserializerTest.java

```diff
@@ -0,0 +1,104 @@
+        JsonParseException exception = assertThrows(
+                JsonParseException.class,
+                () -> mangaDeserializer.deserialize(null, Manga.class, null)
+        );
+        assertEquals("corrupt Manga object", exception.getMessage());
+    }
+
+    @Test
+    void deserialize_nonJsonObject_throwsJsonParseException() {
+        JsonElement jsonElement = JsonParser.parseString("\"Not an object\"");
+        JsonParseException exception = assertThrows(
+                JsonParseException.class,
+                () -> mangaDeserializer.deserialize(jsonElement, Manga.class, null)
+        );
+        assertEquals("corrupt Manga object", exception.getMessage());
+    }
+
+    @Test
+    void deserialize_missingMangaName_throwsJsonParseException() {
+        JsonObject mangaJson = new JsonObject();
+        JsonParseException exception = assertThrows(
+                JsonParseException.class,
+                () -> mangaDeserializer.deserialize(mangaJson, Manga.class, null)
+        );
+        assertEquals("corrupt manga name", exception.getMessage());
+    }
+
+    @Test
+    void deserialize_invalidMangaNameTypeInt_throwsJsonParseException() {
+        JsonObject authorJson = new JsonObject();
+        authorJson.addProperty("mangaName", 1);  // invalid type, expect String
+        JsonParseException exception = assertThrows(
+                JsonParseException.class,
+                () -> mangaDeserializer.deserialize(authorJson, Manga.class, null)
+        );
+        assertEquals("corrupt manga name", exception.getMessage());
+    }
+
+    @Test
+    void deserialize_missingDeadline_throwsJsonParseException() {
+        JsonObject mangaJson = new JsonObject();
+        mangaJson.addProperty("mangaName", "valid manga name"); // isolate deadline
+        JsonParseException exception = assertThrows(
+                JsonParseException.class,
+                () -> mangaDeserializer.deserialize(mangaJson, Manga.class, null)
+        );
+        assertEquals("corrupt deadline", exception.getMessage());
+    }
+
+    @Test
+    void deserialize_invalidDeadlineTypeInt_throwsJsonParseException() {
+        JsonObject mangaJson = new JsonObject();
+        mangaJson.addProperty("mangaName", "valid manga name"); // isolate deadline
+        mangaJson.addProperty("deadline", 1);  // invalid type, expect String
+        JsonParseException exception = assertThrows(
+                JsonParseException.class,
+                () -> mangaDeserializer.deserialize(mangaJson, Manga.class, null)
+        );
+        assertEquals("corrupt deadline", exception.getMessage());
+    }
+
+    @Test
+    void deserialize_validMangaObject_returnsManga() {
+        Author expectedAuthor = new Author("test");
+        Manga expectedManga = new Manga("manga1", expectedAuthor, "deadline1");
+
+        JsonObject mangaJson = new JsonObject();
+        mangaJson.addProperty("mangaName", "manga1");
+        mangaJson.addProperty("deadline", "deadline1");
```

Comment:
With the new addition of sales data, certain test cases may need to be updated to align with those changes.